src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -843,13 +845,44 @@ public void testPrivateKeyInConnectionString() throws SQLException, IOException
           (int) ErrorCode.INVALID_OR_UNSUPPORTED_PRIVATE_KEY.getMessageCode(), e.getErrorCode());
     }
     connection.close();
+  }
 
-    // clean up
-    connection = getConnection();
-    statement = connection.createStatement();
-    statement.execute("use role accountadmin");
-    statement.execute(String.format("alter user %s unset rsa_public_key", testUser));
-    connection.close();
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testPrivateKeyInConnectionStringWithBouncyCastle() throws SQLException, IOException {
```

Comment:
let's add a comment above this test that it will work in snowflake-jdbc version above 3.15.1